website/client/src/components/chat/autoComplete.vue

```diff
@@ -128,38 +130,29 @@ export default {
         backgroundColor: 'white',
       };
     },
-    searchResults () {
-      if (!this.searchActive) return [];
-      if (!this.atRegex.exec(this.text)) return [];
-      this.currentSearch = this.atRegex.exec(this.text)[0]; // eslint-disable-line vue/no-side-effects-in-computed-properties, max-len, prefer-destructuring
-      this.currentSearch = this.currentSearch.substring(1, this.currentSearch.length); // eslint-disable-line vue/no-side-effects-in-computed-properties, max-len
-
-      const lowerCaseSearch = this.currentSearch.toLowerCase();
-      return this.tmpSelections
-        .filter(option => { // eslint-disable-line arrow-body-style
-          return option.displayName.toLowerCase().indexOf(lowerCaseSearch) !== -1
-            || (option.username
-              && option.username.toLowerCase().indexOf(lowerCaseSearch) !== -1);
-        })
-        .slice(0, 4);
-    },
-
   },
   watch: {
-    text (newText) {
-      if (!newText[newText.length - 1] || newText[newText.length - 1] === ' ') {
+    text (newTextParam, prevTextParam) {
+      const delCharFocusBool = prevTextParam.length > newTextParam.length;
+      const caretPosition = this.textbox.selectionEnd;
+      const charFocus = (
+        delCharFocusBool ? prevTextParam : newTextParam
+      )[caretPosition - (delCharFocusBool ? 0 : 1)];
```

Comment:
I think this line would be clearer if you stick to a single ternary (And I'd recommend renaming `charFocus` to something like `lastFocussedChar`)

```suggestion
      const lastFocussedChar = isDeleting ? prevText[caretPosition] : newText[caretPosition - 1];
```

Reply:
Ah yeah, this was relic code that I had when I stored to focusText (including the focusChar)

Which is why the double ternary still exists.

Good suggestion.